src/backend/InvenTree/stock/tasks.py

```diff
@@ -7,23 +7,53 @@
 logger = structlog.get_logger('inventree')
 
 
-@tracer.start_as_current_span('rebuild_stock_item_tree')
-def rebuild_stock_item_tree(tree_id=None):
+@tracer.start_as_current_span('rebuild_stock_items')
+def rebuild_stock_items():
+    """Rebuild the entire StockItem tree structure.
+
+    This may be necessary if the tree structure has become corrupted or inconsistent.
+    """
+    from InvenTree.exceptions import log_error
+    from stock.models import StockItem
+
+    logger.info('Rebuilding StockItem tree structure')
+
+    try:
+        StockItem.objects.rebuild()
+    except Exception as e:
+        log_error('rebuild_stock_items')
+        logger.exception('Failed to rebuild StockItem tree: %s', e)
+
+
+def rebuild_stock_item_tree(tree_id: int, rebuild_on_fail: bool = True) -> bool:
     """Rebuild the stock tree structure.
 
-    The StockItem tree uses the MPTT library to manage the tree structure.
+    Arguments:
+        tree_id (int): The ID of the StockItem tree to rebuild.
+        rebuild_on_fail (bool): If True, will attempt to rebuild the entire StockItem tree if the partial rebuild fails.
+
+    Returns:
+        bool: True if the partial tree rebuild was successful, False otherwise.
+
+    - If the rebuild fails, schedule a rebuild of the entire StockItem tree.
```

Comment:
C: This would be a good metric to track on the demo instance

Reply:
I've just added sentry reporting for errors in tree rebuild